### PR TITLE
fix(dev-infra): use API pagination for retrieving project branches 

### PR DESCRIPTION
**Pull request by devversion**
See individual commits.

### CODE REVIEW

**Comment by devversion** [comment]
cc. @alxhub.

File: dev-infra/utils/testing/github-pagination-header.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/**
+ * Builds the `Link` response header to indicate an API response that is suitable
+ * for pagination. This follows the specification as outlined within:
+ * https://docs.github.com/en/rest/guides/traversing-with-pagination
+ */
+export function buildGithubPaginationResponseHeader(
+    amountPages: number, currentPage: number, baseUrl: string) {
```
Comment (josephperrott) [inline dev-infra/utils/testing/github-pagination-header.ts:15]: maybe `totalPages` instead of `amountPages`?

**Review by josephperrott — APPROVED** [review]
Nice update of octokit too!

**Comment by jessicajaniuk** [comment]
@devversion This PR has conflicts with `12.1.x`. I've switched this to `target: minor` and will land it there. If you'd like this to land in the patch branch, please create a patch PR.

**Pull request closed without merging**